### PR TITLE
Update SNR versions

### PR DESCRIPTION
**Pull request by mshitrit**
**First** Commit:
Go 1.18
operator-sdk 1.19.0
kustomize 4.5.4
controller-gen 0.8.0
envtest k8s 1.23
**Second** Commit:
go mod vendor

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,11 +93,8 @@ verify-no-changes: ## verify there are no un-staged changes
 fetch-mutation: ## fetch mutation package.
 	GO111MODULE=off go get -t -v github.com/mshitrit/go-mutesting/...
 
-ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 test: manifests generate fmt vet ## Run tests.
-	mkdir -p ${ENVTEST_ASSETS_DIR}
-	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.7.2/hack/setup-envtest.sh
-	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./api/... ./controllers/... ./pkg/... -test.v -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/testbin)" go test ./controllers/... -coverprofile cover.out -v -ginkgo.v
```
Comment (slintes) [inline Makefile:97]: you probably want to add the `pkg` package to the packages to test

File: Makefile [hunk 2]
```diff
@@ -138,24 +135,28 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
...
 @[ -f $(1) ] || { \
 set -e ;\
 TMP_DIR=$$(mktemp -d) ;\
 cd $$TMP_DIR ;\
 go mod init tmp ;\
-BIN_DIR=$$(dirname $(1)) ;\
-mkdir -p $$BIN_DIR ;\
 echo "Downloading $(2)" ;\
-GOBIN=$$BIN_DIR go get $(2) ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
```
Comment (slintes) [inline Makefile:158]: the `go get` line be deleted

**Review by slintes — CHANGES_REQUESTED** [review]

File: .github/workflows/post-submit.yaml [hunk 3]
```diff
@@ -19,7 +19,12 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-          
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
```
Comment (razo7) [inline .github/workflows/post-submit.yaml:24]: I think it should be version 3 (see [here](https://github.com/actions/setup-go)).

File: Makefile (hunk 1 above)
Comment (slintes) [reply Makefile]: @mshitrit Please also add the api package, so we will run the soon-to-be-merged new webhook tests as well.
@shellykatz fyi

**Review by slintes — APPROVED** [review]

**Pull request merged**